src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -8143,6 +8148,23 @@ private static boolean handleCosmicBowlingBall(String text) {
     return false;
   }
 
+  private static final Pattern TRAINSET_MOVE =
+      Pattern.compile("^Your toy train moves ahead to the (.+?)\\.");
+
+  private static boolean handleTrainset(TagNode node) {
+    String text = node.getText().toString();
+
+    Matcher matcher = TRAINSET_MOVE.matcher(node.getText().toString());
```

Comment:
should probably reuse text here instead of getting the text again?

Also can you not pass in `str` from the caller instead of passing in the node again?

Reply:
Good catch, and no. The `str` has tags stripped out as far as I can see. Only the actual text in the html node is given.